### ISSUE TITLE
SPM fails to install multiple packages

### ISSUE THREAD

**Issue by gmacon** [saltstack/salt#42646]
### Description of Issue/Question

With the upgrade to Salt 2017.7.0, installing multiple SPM packages at once fails with an uncaught exception. Installing one package at a time still works OK.

### Setup

SPM should be configured to point to a repo with multiple packages.

### Steps to Reproduce Issue

Running (assuming that `foo` and `bar` packages exist in the repo):

```sh
spm install -y -f foo bar
```

produces this error:

```
[ERROR   ] An un-handled exception was caught by salt's global exception handler:
KeyError: "filename 'foo/FORMULA' not found"
Traceback (most recent call last):
  File "/usr/bin/spm", line 12, in <module>
    salt_spm()
  File "/usr/lib/python2.7/site-packages/salt/scripts.py", line 487, in salt_spm
    spm.run()
  File "/usr/lib/python2.7/site-packages/salt/cli/spm.py", line 34, in run
    client.run(self.args)
  File "/usr/lib/python2.7/site-packages/salt/spm/__init__.py", line 113, in run
    self._install(args)
  File "/usr/lib/python2.7/site-packages/salt/spm/__init__.py", line 363, in _install
    self._install_indv_pkg(package, out_file)
  File "/usr/lib/python2.7/site-packages/salt/spm/__init__.py", line 455, in _install_indv_pkg
    formula_ref = formula_tar.extractfile('{0}/FORMULA'.format(pkg_name))
  File "/usr/lib64/python2.7/tarfile.py", line 2111, in extractfile
    tarinfo = self.getmember(member)
  File "/usr/lib64/python2.7/tarfile.py", line 1796, in getmember
    raise KeyError("filename %r not found" % name)
KeyError: "filename 'foo/FORMULA' not found"
```

Though exactly which file is reported as "not found" is consistent for the same list of packages, it's not always the first one.

### Versions Report

```
Salt Version:
           Salt: 2017.7.0

Dependency Versions:
           cffi: 1.6.0
       cherrypy: Not Installed
       dateutil: 1.5
      docker-py: Not Installed
          gitdb: Not Installed
      gitpython: Not Installed
          ioflo: Not Installed
         Jinja2: 2.7.2
        libgit2: Not Installed
        libnacl: Not Installed
       M2Crypto: 0.21.1
           Mako: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 0.4.8
   mysql-python: Not Installed
      pycparser: 2.14
       pycrypto: 2.6.1
   pycryptodome: 3.4.3
         pygit2: Not Installed
         Python: 2.7.5 (default, Aug  2 2016, 04:20:16)
   python-gnupg: Not Installed
         PyYAML: 3.11
          PyZMQ: 15.3.0
           RAET: Not Installed
          smmap: Not Installed
        timelib: Not Installed
        Tornado: 4.2.1
            ZMQ: 4.1.4

System Versions:
           dist: redhat 7.3 Maipo
         locale: UTF-8
        machine: x86_64
        release: 3.10.0-514.el7.x86_64
         system: Linux
        version: Red Hat Enterprise Linux Server 7.3 Maipo
```

**Comment by gtmanfred** [comment]
Was it possible to install more than one package on older releases of salt?

If not, I am going to mark this as a feature request instead of as a bug.

Thanks,
Daniel

**Comment by gmacon** [comment]
It was possible on the previous version. I noticed this when our automated deployment infrastructure broke. I've worked around the immediate issue with `xargs -n1`.

**Comment by gtmanfred** [comment]
Which version did this work in just so I can test and do a git bisect to find where the problem was introduced?

Thanks!
Daniel

**Comment by gmacon** [comment]
Looks like we were previously on 2016.11.6.

**Comment by gtmanfred** [comment]
This bug was introduced in
http://github.com/saltstack/salt/commit/d1f0a74a7cb7257ecf71535361f999229ffbc753

**Comment by Ch3LL** [comment]
@gtmanfred assigned to you for the 2017.7.2 release Thanks

**Comment by gtmanfred** [comment]
This is fixed in #42778
